pkg/http/api_directmessaging.go

```diff
@@ -143,9 +143,14 @@ func (a *api) onDirectMessage(w http.ResponseWriter, r *http.Request) {
 				resStatus.Code = statusCode
 			}
 		} else if resStatus.Code < 200 || resStatus.Code > 399 {
```

Comment:
Not a comment on the PR particularly: But, I want to understand dapr resiliency behavior in general: Here, we see that 4xx error are retried, wheras most of them are non-retriable (1 quick ref: https://stackoverflow.com/a/47841642/18848251). Is it correct to retry errors like 404?

Reply:
I think this is something that needs to be discussed separate from this PR. Can you create an issue regarding this?

Reply:
I agree with @DeepanshuA this may be worth a conversation. There is also an issue that involves being "smarter" about retries depending on status codes (such as respecting Retry-after in 427 responses: #5872 )

Also agree with @mukundansundar it should be done in a separate issue.